src/pkg/packager/deprecated/common.go

```diff
@@ -39,25 +39,33 @@ var breakingChanges = []BreakingChange{
 
 // MigrateComponent runs all migrations on a component.
 // Build should be empty on package create, but include just in case someone copied a zarf.yaml from a zarf package.
-func MigrateComponent(build types.ZarfBuildData, c types.ZarfComponent) types.ZarfComponent {
+func MigrateComponent(build types.ZarfBuildData, c types.ZarfComponent) (types.ZarfComponent, []string) {
+	var packageWarnings = []string{}
```

Comment:
I think this is a bit much for what is a string under the hood - it will require a wrapper every time we want to print them or make them too.

I did make it a named return though so that *should* show up for most people in intellilsense without the need for wrappers
![image](https://github.com/defenseunicorns/zarf/assets/3977569/d1db1deb-e62c-4a5d-bd4a-6e22ab3b438b)